### PR TITLE
[New Sample] - Bot-Tag-Mention -  Csharp

### PR DESCRIPTION
**Pull request by Prithvi-MSFT**

### CODE REVIEW

**Comment by Jegadeesh-MSFT** [comment]
Build Pipeline 

https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/6107799943

image

File: samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+                        // Pull in the data from the Microsoft Graph.
+                        var client = new SimpleGraphClient(tokenResponse.Token);
+                        var teamID = await TeamsInfo.GetTeamDetailsAsync(stepContext.Context, stepContext.Context.Activity.TeamsGetTeamInfo().Id, cancellationToken);
+                        var result = await client.GetTag(teamID.AadGroupId);
+                        var adaptiveCardTemplate = Path.Combine(".", "Resources", "UserMentionCardTemplate.json");
+                        var templateJSON = File.ReadAllText(adaptiveCardTemplate);
+                        AdaptiveCardTemplate template = new AdaptiveCardTemplate(templateJSON);
+                        var memberData = new
+                        {
+                            tagId = result.CurrentPage[0].Id,
```
Comment (limamicro) [inline samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs:81]: can you add the logic to check whether there is existing tag in current team before using CurrentPage[0]?

File: samples/bot-tag-mention/csharp/README.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    - **Edit** the `manifest.json` for `validDomains` with base Url domain. E.g. if you are using ngrok it would be `https://1234.ngrok-free.app` then your domain-name will be `1234.ngrok-free.app` and if you are using dev tunnels, your URL will be like: https://12345.devtunnels.ms.
+    - **Zip** up the contents of the `TeamsAppManifest` folder to create a `manifest.zip` (Make sure that zip file does not contains any subfolder otherwise you will get error while uploading your .zip package)
+    - **Upload** the `manifest.zip` to Teams (In Teams Apps/Manage your apps click "Upload an app". Browse to and Open the .zip file. At the next dialog, click the Add button.)
+    - Add the app to personal/team/groupChat scope (Supported scopes)
+
+**Note**: If you are facing any issue in your app, please uncomment [this](https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/main/samples/Tag-Mention-Bot/csharp/AdapterWithErrorHandler.cs#L25) line and put your debugger for local debug.
+
+
+## Running the sample
```
Comment (limamicro) [inline samples/bot-tag-mention/csharp/README.md:90]: Shall we add one step to tell bot developers to create tags in team before using the sample bot?

**Review by limamicro — COMMENTED** [review]
The overflow flow looks good for me, and I add some nit comments.
Right now in this example, bot developer get tag info only via Graph call. I got several requests from In-store app developers that they do not have permission to call Graph API, and the alternative way to achieve tag mention is by mentioning the tag in the message sent to bot, and the tag info will appear under entities array in the payload.
Therefore, I am thinking that shall we add an alternative way for developers to get tag info if they cannot get it via Graph call?

File: samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs (hunk 1 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs]: Added changes to the code with all necessary exception handling features.

File: samples/bot-tag-mention/csharp/README.md (hunk 2 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/README.md:91]: I have added it in Prerequisites and the bot will send error message if they try to access it without creating a tag.

File: samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+            bool tagExists = false;
+            if (tokenResponse?.Token != null)
+            {
+                try
+                {
+                    stepContext.Context.Activity.RemoveRecipientMention();
+                    if (stepContext.Context.Activity.Text.Trim().ToLower().Contains("<at>"))
+                    {
+                        var tagName = stepContext.Context.Activity.Text.Replace("<at>", string.Empty).Replace("</at>", string.Empty).Trim();
+                        var tagID = stepContext.Context.Activity.Entities[1].Properties.First.First.First.First.ToString();
```
Comment (limamicro) [inline samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs:92]: A more straightforward way to get id is:
tagId = stepContext.Context.Activity.Entities[1].Properties["mentioned"]["id"]ToString();

**Review by limamicro — CHANGES_REQUESTED** [review]

File: samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs (hunk 3 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs]: Thanks for mentioning it, I have fixed it now.

**Review by limamicro — APPROVED** [review]

File: .github/workflows/build-complete-samples.yml [hunk 4]
```diff
@@ -440,6 +440,10 @@ jobs:
               name: 'meetings-app-icon-badging'
               version: '6.0.x'
 
+            - project_path: 'samples/bot-tag-mention/csharp/TagMentionBot.csproj'
```
[the remark this answers is not in the document]
Comment (ChetanSharma-msft) [reply .github/workflows/build-complete-samples.yml:443]: And run build pipeline on this branch.

File: samples/bot-tag-mention/csharp/AdapterWithErrorHandler.cs [hunk 5]
```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Bot.Builder.Integration.AspNet.Core;
+using Microsoft.Bot.Builder.TraceExtensions;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.BotBuilderSamples
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/AdapterWithErrorHandler.cs:9]: Wrong namespace. Please try to create an new .cs file and check it's namespace.

File: samples/bot-tag-mention/csharp/AppManifest/manifest.json [hunk 6]
```diff
@@ -0,0 +1,47 @@
+{
+  "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.14/MicrosoftTeams.schema.json",
+  "manifestVersion": "1.14",
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/AppManifest/manifest.json:3]: Please check if we can use latest version of manifest.

File: samples/bot-tag-mention/csharp/Bots/DialogBot.cs [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Builder.Teams;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.BotBuilderSamples
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/Bots/DialogBot.cs:13]: Wrong namespace

File: samples/bot-tag-mention/csharp/Bots/TeamsConversationBot.cs [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Schema;
+using Microsoft.Bot.Schema.Teams;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.BotBuilderSamples.Bots
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/Bots/TeamsConversationBot.cs:14]: Wrong namespace

File: samples/bot-tag-mention/csharp/Bots/TeamsConversationBot.cs [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+    {
+        private string _appId;
+        private string _appPassword;
+
+        public TeamsConversationBot(ConversationState conversationState, UserState userState, T dialog, ILogger<DialogBot<T>> logger)
+                    : base(conversationState, userState, dialog, logger)
+        {
+        }
+
+        protected override async Task OnTeamsMembersAddedAsync(IList<TeamsChannelAccount> membersAdded, TeamInfo teamInfo, ITurnContext<IConversationUpdateActivity> turnContext, CancellationToken cancellationToken)
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/Bots/TeamsConversationBot.cs:26]: Add methods summary.

File: samples/bot-tag-mention/csharp/Bots/TeamsConversationBot.cs [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+            foreach (var teamMember in membersAdded)
+            {
+                if (teamMember.Id != turnContext.Activity.Recipient.Id && turnContext.Activity.Conversation.ConversationType != "personal")
+                {
+                    await turnContext.SendActivityAsync(MessageFactory.Text($"Welcome to the team {teamMember.GivenName} {teamMember.Surname}."), cancellationToken);
+                }
+            }
+        }
+
+        protected override async Task OnInstallationUpdateActivityAsync(ITurnContext<IInstallationUpdateActivity> turnContext, CancellationToken cancellationToken)
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/Bots/TeamsConversationBot.cs:37]: Add method summary

File: samples/bot-tag-mention/csharp/Controllers/BotController.cs [hunk 11]
```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Integration.AspNet.Core;
+
+namespace Microsoft.BotBuilderSamples.Controllers
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/Controllers/BotController.cs:9]: Wrong namespace

File: samples/bot-tag-mention/csharp/Dialogs/LogoutDialog.cs [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+// </copyright>
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.BotBuilderSamples
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/Dialogs/LogoutDialog.cs:12]: Wrong namespace

File: samples/bot-tag-mention/csharp/Dialogs/LogoutDialog.cs [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+            return await base.OnBeginDialogAsync(innerDc, options, cancellationToken);
+        }
+
+        /// <summary>
+        /// Called when the dialog is _continued_, where it is the active dialog and the user replies with a new activity.
+        /// </summary>
+        /// <param name="innerDc">The inner DialogContext for the current turn of conversation.</param>
+        /// <param name="cancellationToken">Propagates notification that operations should be canceled.</param>
+        /// <returns>A task representing the asynchronous operation.</returns>
+        protected override async Task<DialogTurnResult> OnContinueDialogAsync(
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/Dialogs/LogoutDialog.cs:54]: Are we using all of these dialogs?

File: samples/bot-tag-mention/csharp/Dialogs/LogoutDialog.cs [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+        }
+
+        /// <summary>
+        /// Called when the dialog is interrupted, where it is the active dialog and the user replies with a new activity.
+        /// </summary>
+        /// <param name="innerDc">The inner DialogContext for the current turn of conversation.</param>
+        /// <param name="cancellationToken">Propagates notification that operations should be canceled.</param>
+        /// <returns></returns>
+        private async Task<DialogTurnResult> InterruptAsync(
+            DialogContext innerDc,
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/Dialogs/LogoutDialog.cs:74]: Where are we using it?

File: samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs [hunk 15]
```diff
@@ -0,0 +1,143 @@
...
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Builder.Teams;
+using Microsoft.Bot.Schema;
+using Microsoft.Bot.Schema.Teams;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Microsoft.BotBuilderSamples
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs:19]: Wrong namespace

File: samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs [hunk 16]
```diff
@@ -0,0 +1,143 @@
...
+    {
+        protected readonly ILogger _logger;
+
+        public MainDialog(IConfiguration configuration, ILogger<MainDialog> logger)
+            : base(nameof(MainDialog), configuration["ConnectionName"])
+        {
+            _logger = logger;
+
+            AddDialog(new OAuthPrompt(
+                nameof(OAuthPrompt),
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs:31]: Not sure if we are using auth related stuff in this sample.

File: samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs [hunk 17]
```diff
@@ -0,0 +1,143 @@
...
+            {
+                PromptStepAsync,
+                MentionTagAsync
+            }));
+
+            // The initial child Dialog to run.
+            InitialDialogId = nameof(WaterfallDialog);
+        }
+
+        private async Task<DialogTurnResult> PromptStepAsync(WaterfallStepContext stepContext, CancellationToken cancellationToken)
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs:53]: Add method summary.

File: samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs [hunk 18]
```diff
@@ -0,0 +1,143 @@
...
+        {
+            _logger.LogInformation("PromptStepAsync() called.");
+            return await stepContext.BeginDialogAsync(nameof(OAuthPrompt), null, cancellationToken);
+        }
+
+        private async Task<ResourceResponse> TagMentionAdaptivecard(WaterfallStepContext stepContext, CancellationToken cancellationToken, string tagName, string tagId)
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs:59]: Add method summary.

File: samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs [hunk 19]
```diff
@@ -0,0 +1,143 @@
...
+        {
+            var adaptiveCardTemplate = Path.Combine(".", "Resources", "UserMentionCardTemplate.json");
+            var templateJSON = File.ReadAllText(adaptiveCardTemplate);
+            AdaptiveCardTemplate template = new AdaptiveCardTemplate(templateJSON);
+            var memberData = new
+            {
+                tagId = tagId,
+                tagName = tagName
+            };
+            string cardJSON = template.Expand(memberData);
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs:69]: Add one line space

File: samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs [hunk 20]
```diff
@@ -0,0 +1,143 @@
...
+            var adaptiveCardAttachment = new Attachment
+            {
+                ContentType = "application/vnd.microsoft.card.adaptive",
+                Content = JsonConvert.DeserializeObject(cardJSON),
+            };
+            return await stepContext.Context.SendActivityAsync(MessageFactory.Attachment(adaptiveCardAttachment), cancellationToken);
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs:75]: One line space

File: samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs [hunk 21]
```diff
@@ -0,0 +1,143 @@
...
+        }
+
+        private async Task<DialogTurnResult> MentionTagAsync(WaterfallStepContext stepContext, CancellationToken cancellationToken)
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs:78]: Add method summary

File: samples/bot-tag-mention/csharp/Program.cs [hunk 22]
```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.BotBuilderSamples
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/Program.cs:8]: Wrong namespace name

File: samples/bot-tag-mention/csharp/README.md [hunk 23]
```diff
@@ -0,0 +1,138 @@
...
+
+   In Azure portal, create a [Azure Bot resource](https://docs.microsoft.com/azure/bot-service/bot-service-quickstart-registration).
+    - For bot handle, make up a name.
+    - Select "Use existing app registration" (Create the app registration in Azure Active Directory beforehand.)
+    - __*If you don't have an Azure account*__ create an [Azure free account here](https://azure.microsoft.com/free/)
+    
+   In the new Azure Bot resource in the Portal, 
+    - Refer to [Bot SSO Setup document](https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/main/samples/bot-conversation-sso-quickstart/BotSSOSetup.md).
+    - Ensure that you've [enabled the Teams Channel](https://learn.microsoft.com/azure/bot-service/channel-connect-teams?view=azure-bot-service-4.0)
+    - In Settings/Configuration/Messaging endpoint, enter the current `https` URL you were given by running the tunneling application. Append with the path `/api/messages`
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/README.md:62]: I can't see any steps to setup bot authentication and how to configure connection in azure bot.

File: samples/bot-tag-mention/csharp/README.md [hunk 24]
```diff
@@ -0,0 +1,138 @@
...
+products:
+- office-teams
+- office
+- office-365
+languages:
+- csharp
+extensions:
+ contentType: samples
+ createdDate: "09/06/2023 20:38:25 PM"
+urlFragment: officedev-microsoft-teams-samples-Tag-Mention-Bot-csharp
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/README.md:13]: Your actual app name is "bot-tag-mention" but you used wrong names in readme and other places.

File: samples/bot-tag-mention/csharp/README.md [hunk 25]
```diff
@@ -0,0 +1,138 @@
...
+3) Clone the repository
+
+    ```bash
+    git clone https://github.com/OfficeDev/Microsoft-Teams-Samples.git
+    ```
+
+4) If you are using Visual Studio
+   - Launch Visual Studio
+   - File -> Open -> Project/Solution
+   - Navigate to `samples/Tag-Mention-Bot/csharp` folder
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/README.md:73]: This is wrong name

File: samples/bot-tag-mention/csharp/README.md [hunk 26]
```diff
@@ -0,0 +1,138 @@
...
+    - **Edit** the `manifest.json` for `validDomains` with base Url domain. E.g. if you are using ngrok it would be `https://1234.ngrok-free.app` then your domain-name will be `1234.ngrok-free.app` and if you are using dev tunnels, your URL will be like: https://12345.devtunnels.ms.
+    - **Zip** up the contents of the `TeamsAppManifest` folder to create a `manifest.zip` (Make sure that zip file does not contains any subfolder otherwise you will get error while uploading your .zip package)
+    - **Upload** the `manifest.zip` to Teams (In Teams Apps/Manage your apps click "Upload an app". Browse to and Open the .zip file. At the next dialog, click the Add button.)
+    - Add the app to personal/team/groupChat scope (Supported scopes)
+
+**Note**: If you are facing any issue in your app, please uncomment [this](https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/main/samples/Tag-Mention-Bot/csharp/AdapterWithErrorHandler.cs#L25) line and put your debugger for local debug.
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/README.md:88]: This path is also wrong

Comment (ChetanSharma-msft) [reply samples/bot-tag-mention/csharp/README.md:88]: image

File: samples/bot-tag-mention/csharp/README.md [hunk 27]
```diff
@@ -0,0 +1,138 @@
...
+To learn more about deploying a bot to Azure, see [Deploy your bot to Azure](https://aka.ms/azuredeployment) for a complete list of deployment instructions.
+
+## Further reading
+
+- [Bot Framework Documentation](https://docs.botframework.com)
+- [Bot Basics](https://docs.microsoft.com/azure/bot-service/bot-builder-basics?view=azure-bot-service-4.0)
+- [Azure Bot Service Introduction](https://docs.microsoft.com/azure/bot-service/bot-service-overview-introduction?view=azure-bot-service-4.0)
+- [Azure Bot Service Documentation](https://docs.microsoft.com/azure/bot-service/?view=azure-bot-service-4.0)
+- [Messages in bot conversations](https://learn.microsoft.com/microsoftteams/platform/bots/how-to/conversations/conversation-messages?tabs=dotnet)
+- [Tag mention](https://learn.microsoft.com/microsoftteams/platform/bots/how-to/conversations/channel-and-group-conversations?tabs=dotnet#tag-mention)
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/README.md:136]: Put this as first URL.

File: samples/bot-tag-mention/csharp/README.md [hunk 28]
```diff
@@ -0,0 +1,138 @@
...
+
+<img src="https://pnptelemetry.azurewebsites.net/microsoft-teams-samples/samples/Tag-Mention-Bot-csharp" />
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/README.md:138]: This is also wrong name

File: samples/bot-tag-mention/csharp/SimpleGraphClient.cs [hunk 29]
```diff
@@ -0,0 +1,55 @@
+﻿// <copyright file="SimpleGraphClient.cs" company="Microsoft">
+// Copyright (c) Microsoft. All rights reserved.
+// </copyright>
+
+using System;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Microsoft.Graph;
+
+namespace Microsoft.BotBuilderSamples
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/SimpleGraphClient.cs:10]: Wrong namespace

File: samples/bot-tag-mention/csharp/Startup.cs [hunk 30]
```diff
@@ -0,0 +1,75 @@
...
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Integration.AspNet.Core;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.BotBuilderSamples.Bots;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Bot.Connector.Authentication;
+
+namespace Microsoft.BotBuilderSamples
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/Startup.cs:14]: Wrong namespace

File: samples/bot-tag-mention/csharp/appsettings.json [hunk 31]
```diff
@@ -0,0 +1,7 @@
+{
+  "MicrosoftAppType": "Multitenant",
+  "MicrosoftAppId": "",
+  "MicrosoftAppPassword": "",
+  "MicrosoftAppTenantId": "",
+  "ConnectionName": ""
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/appsettings.json:6]: Mention about this connection also in readme file. How to get it.

File: samples/bot-tag-mention/csharp/assets/sample.json [hunk 32]
```diff
@@ -0,0 +1,68 @@
+[
+  {
+    "name": "officedev-microsoft-teams-samples-Tag-Mention-Bot-csharp",
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/assets/sample.json:3]: Wrong name

File: samples/bot-tag-mention/csharp/assets/sample.json [hunk 33]
```diff
@@ -0,0 +1,68 @@
...
+    "source": "officeDev",
+    "title": "Tag Mention Bot using Bot Framework",
+    "shortDescription": "This sample app demonstrates the use of tag mention funtionality in teams scope.",
+    "url": "https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/bot-conversation/csharp",
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/assets/sample.json:7]: Wrong URL and project name

File: samples/bot-tag-mention/csharp/assets/sample.json [hunk 34]
```diff
@@ -0,0 +1,68 @@
...
+      {
+        "key": "TEAMS-FEATURES",
+        "value": "bot, adaptive card"
+      }
+    ],
+    "thumbnails": [
+      {
+        "type": "image",
+        "order": 100,
+        "url": "https://raw.githubusercontent.com/OfficeDev/Microsoft-Teams-Samples/main/samples/Tag-Mention-Bot/csharp/Images/Tag-Mention-Bot.gif",
```
Comment (ChetanSharma-msft) [inline samples/bot-tag-mention/csharp/assets/sample.json:38]: Wrong file path

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Added my review comments

File: samples/bot-tag-mention/csharp/AdapterWithErrorHandler.cs (hunk 5 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/AdapterWithErrorHandler.cs:9]: Fixed

File: samples/bot-tag-mention/csharp/AppManifest/manifest.json (hunk 6 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/AppManifest/manifest.json:3]: fixed

File: samples/bot-tag-mention/csharp/Bots/DialogBot.cs (hunk 7 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/Bots/DialogBot.cs:13]: fixed

File: samples/bot-tag-mention/csharp/Bots/TeamsConversationBot.cs (hunk 8 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/Bots/TeamsConversationBot.cs:14]: fixed

File: samples/bot-tag-mention/csharp/Bots/TeamsConversationBot.cs (hunk 9 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/Bots/TeamsConversationBot.cs:26]: Fixed

File: samples/bot-tag-mention/csharp/Bots/TeamsConversationBot.cs (hunk 10 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/Bots/TeamsConversationBot.cs:37]: Fixed

File: samples/bot-tag-mention/csharp/Controllers/BotController.cs (hunk 11 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/Controllers/BotController.cs:9]: Fixed

File: samples/bot-tag-mention/csharp/Dialogs/LogoutDialog.cs (hunk 12 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/Dialogs/LogoutDialog.cs:12]: Fixed

File: samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs (hunk 15 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs:19]: Fixed

File: samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs (hunk 16 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs:31]: Yes, we are using for graph client.

File: samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs (hunk 20 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs:75]: Fixed

File: samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs (hunk 19 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs:69]: Fixed

File: samples/bot-tag-mention/csharp/Program.cs (hunk 22 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/Program.cs:8]: Fixed

File: samples/bot-tag-mention/csharp/README.md (hunk 23 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/README.md:62]: https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/main/samples/bot-conversation-sso-quickstart/BotSSOSetup.md


I have added the above link and it consists everything

File: samples/bot-tag-mention/csharp/README.md (hunk 24 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/README.md:13]: Fixed

File: samples/bot-tag-mention/csharp/README.md (hunk 25 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/README.md:73]: Fixed

File: samples/bot-tag-mention/csharp/README.md (hunk 26 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/README.md:88]: Fixed

File: samples/bot-tag-mention/csharp/README.md (hunk 27 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/README.md:136]: Fixed

File: samples/bot-tag-mention/csharp/README.md (hunk 28 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/README.md:138]: Fixed

File: samples/bot-tag-mention/csharp/SimpleGraphClient.cs (hunk 29 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/SimpleGraphClient.cs:10]: Fixed

File: samples/bot-tag-mention/csharp/Startup.cs (hunk 30 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/Startup.cs:14]: Fixed

File: samples/bot-tag-mention/csharp/assets/sample.json (hunk 32 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/assets/sample.json:3]: Fixed

File: samples/bot-tag-mention/csharp/assets/sample.json (hunk 33 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/assets/sample.json:7]: Fixed

File: samples/bot-tag-mention/csharp/assets/sample.json (hunk 34 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/assets/sample.json:38]: Fixed

File: samples/bot-tag-mention/csharp/Dialogs/LogoutDialog.cs (hunk 13 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/Dialogs/LogoutDialog.cs:54]: Not needed . removed.

File: samples/bot-tag-mention/csharp/appsettings.json (hunk 31 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/appsettings.json:6]: Fixed

File: samples/bot-tag-mention/csharp/Dialogs/LogoutDialog.cs (hunk 14 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/Dialogs/LogoutDialog.cs:55]: Fixed

File: samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs (hunk 17 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs:54]: Fixed

File: samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs (hunk 21 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs:83]: Fixed

File: samples/bot-tag-mention/csharp/README.md (hunk 23 above)
Comment (Jegadeesh-MSFT) [reply samples/bot-tag-mention/csharp/README.md]: Changed the readme accordingly. Fixed

**Review by Pawank-MSFT — APPROVED** [review]
looks correct, Approving!

**Comment by Pawank-MSFT** [comment]
@Jegadeesh-MSFT  Resolved the conflicts.

**Comment by Jegadeesh-MSFT** [comment]
**Executed pipeline**
image
https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/6702737566

**Review by Harikrishnan-MSFT — APPROVED** [review]

**Comment by Pawank-MSFT** [comment]
Execute build pipeline
https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/6732221023

**Review by ChetanSharma-msft — APPROVED** [review]
Changes looks fine, approving it.

**Pull request merged**